src/lib/gundb/UserStorageClass.js

```diff
@@ -1549,7 +1550,7 @@ export class UserStorage {
 
     data.initiatorType = isMobilePhone(data.initiator) ? 'mobile' : isEmail(data.initiator) ? 'email' : undefined
     data.address = data.address && UserStorage.cleanFieldForIndex('walletAddress', data.address)
-    data.value = (receiptData && receiptData.value) || amount
+    data.value = (receiptData && (receiptData.value || receiptData.amount)) || amount
```

Comment:
in receipt i think it is always value, whats the case where it is amount?
there are two places to get the value, either from event details as saved when first enqueued, because user typed how much he want to send so we know the amount before hand or from the blockchain tx receipt
so this is not the correct backup solution

Reply:
@sirpy not always, I checked - if the event wasn't saved (enqueued) for some reason then the amount from tx will be saved exactly in 'event.data.receiptData.amount'.

Reply:
when is it value and when is it amount? it sounds like a bug. it should be called the same always. check where we set it as amount and where as value

Reply:
@sirpy we have amount field saved inside receiptData in a case when enqueueTx failed for withdraw by send link, so event doesn't exist in the gun. Then the event from blockchain handled by handleReceiptUpdated and it saves the arrived tx data to the receiptData obj which contain amount (look at screenshot).
<img width="1560" alt="1" src="https://user-images.githubusercontent.com/49862004/74420017-72976d00-4e53-11ea-9b4e-cf65b2e1cee9.png">

So we are not controlling it in that case. What should we do with it?

Reply:
1. this field amount is in DATA not receipt.
2. we put all the information to transaction, of course we control everything, also the incoming receipt from blockchain we control how it is saved and what are the field names.
so i ask again to check yourself and check the code if we need both amount+value
1. if we need both then we need to change code so there is a single name, there's no reason why we have to fields for the same data
2. we should have the tx amount from two places 1. when the user create the tx so we know the amount 2. from blockchain receipt this should be your fallbacks

Reply:
@sirpy 
The 'Transfer' event from blockchain came with 'value' field which contains the amount of tx.
The 'PaymentWithdraw' event from blockchain came 'amount' field which contains the amount of tx.

That is why we have two different fields.
In our case when enqueueTx failed the 'amount' and the event obj wasn't saved to gun and data was lost. So when the 'PaymentWithdraw' event came from blockchain - it contains 'amount' not 'value' field and saved as receiptData.

So what we can do:
- in case if there is an 'amount' field exists - rename it on 'value'

Reply:
ok now it makes sense